org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/DirtyStateEditorSupport.java

```diff
@@ -479,13 +479,16 @@ public void markEditorClean(IDirtyStateEditorSupportClient client) {
 	
 	@Override
 	public void descriptionsChanged(final IResourceDescription.Event event) {
-		if (!getDirtyResource().isInitialized())
-			return;
-		for(IResourceDescription.Delta delta: event.getDeltas()) {
-			if (delta.getOld() == getDirtyResource().getDescription() || delta.getNew() == getDirtyResource().getDescription())
+		DocumentBasedDirtyResource resource = getDirtyResource();
+		synchronized(resource) {
+			if (!resource.isInitialized())
 				return;
+			for(IResourceDescription.Delta delta: event.getDeltas()) {
+				if (delta.getOld() == resource.getDescription() || delta.getNew() == resource.getDescription())
```

Comment:
i am not sure if it is better to store resource.getDescription in a variable and end synchronization here

Reply:
Yes, that would make sense indeed. The description should be a copy and it *should* be ok to work with that one in the loop without a sync block.

Reply:
@szarnekow updated pr. ok now?